src/overlays/actors/ovl_En_Fz/z_en_fz.c

```diff
@@ -26,73 +70,812 @@ const ActorInit En_Fz_InitVars = {
+
+    this->actor.colChkInfo.mass = 0xFF;
+
+    this->actor.flags &= ~1;
+
+    this->unusedTimer1 = 0;
+    this->unusedCounter = 0;
+    this->isAlwaysTrue = true;
+    this->isMoving = false;
+    this->isFreezing = false;
+    this->isActive = true;
+    this->isDespawning = false;
+
+    this->actor.speedXZ = 0.0f;
+    this->actor.gravity = 0.0f;
+    this->actor.velocity.y = 0.0f;
+
+    this->posOrigin.y = this->actor.posRot.pos.y;
+    this->IceSmokeFreezingSpawnHeight = this->actor.posRot.pos.y;
+
+    this->posOrigin.x = this->actor.posRot.pos.x;
+    this->posOrigin.z = this->actor.posRot.pos.z;
+
+    this->unusedFloat = 135.0f;
+
+    if (this->actor.params < 0) {
+        this->envAlpha = 0;
+        this->actor.scale.y = 0.0f;
+        EnFz_Hidden(this);
+    } else {
+        this->envAlpha = 255;
+        EnFz_BeginFreezingStationary(this);
+    }
+
+    EnFz_GetCollision(this, globalCtx);
+}
+
+void EnFz_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnFz* this = THIS;
+    Collider_DestroyCylinder(globalCtx, &this->collider1);
+    Collider_DestroyCylinder(globalCtx, &this->collider2);
+    Collider_DestroyCylinder(globalCtx, &this->collider3);
+}
+
+void EnFz_GetCollision(EnFz* this, GlobalContext* globalCtx) {
+    Vec3f pos;
+    Vec3f collisionPos;
+    Vec3f vec1;
+    u32 bgId;
+    CollisionPoly* poly;
+
+    pos.x = this->actor.posRot.pos.x;
+    pos.y = this->actor.posRot.pos.y + 20.0f;
+    pos.z = this->actor.posRot.pos.z;
+    Matrix_Translate(pos.x, pos.y, pos.z, MTXMODE_NEW);
+    Matrix_RotateRPY(this->actor.shape.rot.x, this->actor.shape.rot.y, this->actor.shape.rot.z, MTXMODE_APPLY);
+
+    vec1.y = 0.0f;
+    vec1.x = 0.0f;
+    vec1.z = 220.0f;
+    Matrix_MultVec3f(&vec1, &this->collisionPos);
+
+    // If Collided
+    if (func_8003DE84(&globalCtx->colCtx, &pos, &this->collisionPos, &collisionPos, &poly, 1, 0, 0, 1, &bgId)) {
+        Math_Vec3f_Copy(&this->collisionPos, &collisionPos);
+    }
+
+    pos.x = this->actor.posRot.pos.x - this->collisionPos.x;
+    pos.z = this->actor.posRot.pos.z - this->collisionPos.z;
+    this->xzDistSquare = SQ(pos.x) + SQ(pos.z);
+}
+
+s32 EnFz_IsCollision(EnFz* this, Vec3f* vec) {
+    f32 x_dist;
+    f32 z_dist;
+
+    x_dist = this->actor.posRot.pos.x - vec->x;
+    z_dist = this->actor.posRot.pos.z - vec->z;
+
+    if (this->xzDistSquare <= (SQ(x_dist) + SQ(z_dist))) {
+        return true;
+    }
+    else {
+        return false;
+    }
+}
+
+void EnFz_Damaged(EnFz* this, GlobalContext* globalCtx, Vec3f* vec, s32 numEffects, f32 unk_float) {
+    int i;
+    Vec3f pos;
+    Vec3f vel;
+    Vec3f accel;
+    Color_RGBA8 primColor;
+    Color_RGBA8 envColor;
+    f32 scale;
+    s32 life;
+
+    primColor.r = 155;
+    primColor.g = 255;
+    primColor.b = 255;
+    primColor.a = 255;
+    envColor.r = 200;
+    envColor.g = 200;
+    envColor.b = 200;
+    accel.z = 0.0f;
+    accel.y = -1.0f;
+    accel.x = 0.0f;
```

Comment:
```suggestion
    accel.x = accel.z = 0.0f;
    accel.y = -1.0f;
```
Assuming it matches.